src/components/Modal/FormModal.tsx

```diff
@@ -1,38 +1,122 @@
+import React, {useState} from 'react'
 import { Modal } from '@mui/material'
 import { Box } from '@mui/system'
 import Button from '@mui/material/Button'
+import {Wine, WineData} from '../../interfaces/wine_interface'
 
 export const FormModal: React.FC<{
   openAddModal: boolean,
   handleCloseAddModal: () => void
   header: string
-}> = ({ openAddModal, handleCloseAddModal, header }) => {
+  createWine: (wineData: WineData) => Promise<any>
+  updateWinesList: (wine: Wine) => void
+}> = ({ openAddModal, handleCloseAddModal, header, createWine, updateWinesList }) => {
+
+  const [wineValues, setWineValues] = useState<WineData>({
+    name: "",
+    description: "",
+    rating: 0,
+    quantity: 0
+  })
+
+  const [message, setMessage] = useState("")
+  const [errors, setErrors] = useState<string[]>([])
+
+  const onChange: React.ChangeEventHandler<HTMLInputElement> = e => {
+    setWineValues({...wineValues, [e.target.name]: e.target.value})
+  }
+
+  const handleSubmit = async (e: React.SyntheticEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    await createWine(wineValues).then(response => {
+      updateWinesList(response.data.attributes)
+      setWineValues({ name: "", description: "", rating: 0, quantity: 0 })
+      setMessage("Successfully added a wine!")
+    })
+      .catch(error => {
+        setMessage("")
+        const errorData = error.response.data.errors
+        const formattedErrors = formatErrors(errorData)
+        setErrors(formattedErrors)
+    })
+  }
+
+  const formatErrors = (errors: any) => {
+    const errorKeys = Object.keys(errors)
+
+    const formattedErrors = errorKeys.map((errorKey: string) => `${errorKey} ${errors[errorKey]}`)
+
+    return formattedErrors
+  }
```

Comment:
Looks like this method doesn't need to be inside this component. Moving it out can have some benefits: reuse of function throughout the code, and easier to read.